deebot_client/messages/json/map.py

```diff
@@ -33,3 +58,50 @@ def _handle_body_data_dict(
         return HandlingResult(
             HandlingState.SUCCESS, {"mid": data["mid"], "type": data["type"]}
         )
+
+
+class OnMapTrace(MessageBodyDataDict):
+    """On map trace message."""
+
+    name = "onMapTrace"
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        if not data.get("traceValue"):
+            # TODO verify that this is legit pylint: disable=fixme
+            return HandlingResult.analyse()
+
+        total = int(data["totalCount"])
+        start = int(data["traceStart"])
+
+        event_bus.notify(
+            MapTraceEvent(start=start, total=total, data=data["traceValue"])
+        )
+        return HandlingResult(HandlingState.SUCCESS, {"start": start, "total": total})
+
+
+class OnMinorMap(MessageBodyDataDict):
+    """On minor map message."""
+
+    name = "onMinorMap"
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        # onMinorMap sends no type, so fallback to "ol"
```

Comment:
From this comment, it looks like `getMinorMap` and `onMinorMap` are sending different data. This function should only have the code for the message. Code for `getMinorMap` should be part of the `GetMinorMap` class